src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-manager-message.service.ts

```diff
@@ -5,204 +5,161 @@ import { FinishedTask } from '../models/finished-task';
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}'`,
-      (metadata) =>
-        `Updating snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}'`,
-      (metadata) =>
-        `Snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}' ` +
-        `has been updated successfully`,
-      () => {
-        return {
-          '16': `Cannot unprotect snapshot because it contains child images.`
-        };
-      }
+      this.commonOperations.update,
+      this.rbd.snapshot,
+      (metadata) => ({
+        '16': `Cannot unprotect ${this.rbd.snapshot(
+          metadata
+        )} because it contains child images.`
+      })
     ),
     'rbd/snap/delete': new TaskManagerMessage(
-      (metadata) =>
-        `Delete snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}'`,
-      (metadata) =>
-        `Deleting snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}'`,
-      (metadata) =>
-        `Snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}' ` +
-        `has been deleted successfully`,
-      () => {
-        return {
-          '16': `Snapshot is protected.`
-        };
-      }
+      this.commonOperations.delete,
+      this.rbd.snapshot,
+      (metadata) => ({
+        '16': `Cannot delete ${this.rbd.snapshot(metadata)} because it's protected.`
+      })
     ),
     'rbd/snap/rollback': new TaskManagerMessage(
-      (metadata) =>
-        `Rollback snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}'`,
-      (metadata) =>
-        `Rolling back snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}'`,
-      (metadata) =>
-        `Snapshot ` +
-        `'${metadata.pool_name}/${metadata.image_name}@${metadata.snapshot_name}' ` +
-        `has been rolled back successfully`,
-      () => {
-        return {};
-      }
+      new TaskMessageOperation('Rolling back', 'rollback', 'Rolled back'),
+      this.rbd.snapshot
     )
   };
 
-  defaultMessage = new TaskManagerMessage(
-    (metadata) => {
-      return Components[metadata.component] || metadata.component || 'Unknown Task';
-    },
-    (metadata) => {
-      return (
-        'Executing ' + (Components[metadata.component] || metadata.component || 'unknown task')
-      );
-    },
-    (metadata) => 'Task executed successfully',
-    () => {
-      return {};
-    }
-  );
-
   constructor() {}
 
-  getSuccessMessage(finishedTask: FinishedTask) {
-    const taskManagerMessage = this.messages[finishedTask.name] || this.defaultMessage;
-    return taskManagerMessage.success(finishedTask.metadata);
+  _getTaskMessage(task: Task) {
+    return this.messages[task.name] || this.defaultMessage;
+  }
+
+  getSuccessMessage(task: FinishedTask) {
```

Comment:
Since we are always using the value of this method as the title in the `show` method, I think this should be rename `getSuccessTitle`.